larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -524,6 +524,49 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
         for (const ParticleFlowObject *const pPfo : pfoList) pfoVector.push_back(pPfo);
         std::sort(pfoVector.begin(), pfoVector.end(), LArPfoHelper::SortByNHits);
 
+        std::pair<const LArTPC*, const LArTPC*> stitchedLArTPCs;
+
+        try
+        {
+            for (const ParticleFlowObject *const pPfoToShift : pfoVector)
+            {
+                if (pfoToLArTPCMap.find(pPfoToShift) != pfoToLArTPCMap.end())
+                {
+                    const LArTPC *const pLArTPC(pfoToLArTPCMap.at(pPfoToShift));
+
+                    if (pLArTPC && (stitchedLArTPCs.first == pLArTPC || stitchedLArTPCs.second == pLArTPC))
+                        continue;
+
+                    if (!stitchedLArTPCs.first)
+                    {
+                        stitchedLArTPCs.first = pLArTPC;
+                    }
+                    else if (!stitchedLArTPCs.second)
+                    {
+                        stitchedLArTPCs.second = pLArTPC;
+                    }
+                    else
+                    {
+                        throw StatusCodeException(STATUS_CODE_FAILURE);
+                    }
+                }
+                else
+                {
+                    throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+                }
+            }
+        }
+        catch (const pandora::StatusCodeException &statusCodeException)
+        {
+            if (STATUS_CODE_FAILURE == statusCodeException.GetStatusCode())
+                std::cout << "StitchingCosmicRayMergingTool: Attempting to stitch a pfo across more than two LArTPCs" << std::endl;
+
+            if (STATUS_CODE_NOT_FOUND == statusCodeException.GetStatusCode())
+                std::cout << "StitchingCosmicRayMergingTool: Unable to find LArTPC for a pfo" << std::endl;
+
+            continue;
```

Comment:
I think it looks like you do indeed want to continue, as you'll keep on trying to stitch pPfoToEnlarge to those Pfos in the pfoVector. (This kind of logic is often a source of problems, but looks good here!)